pkg/registry/apis/dashboard/legacy_storage.go

```diff
@@ -66,7 +81,37 @@ func (s *storeWrapper) Create(ctx context.Context, obj runtime.Object, createVal
 			meta.SetDeprecatedInternalID(access.DashboardID) //nolint:staticcheck
 		}
 	}
-	return obj, err
+
+	if err != nil {
+		return obj, err
+	}
+
+	unstructuredMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		return obj, err
+	}
+	unstructuredObj := &unstructured.Unstructured{Object: unstructuredMap}
+
+	user, err := identity.GetRequester(ctx)
+	if err != nil {
+		return obj, err
+	}
+
+	legacyDashboard, err := (*s.DashboardService).UnstructuredToLegacyDashboard(ctx, unstructuredObj, user.GetOrgID())
+	if err != nil {
+		return obj, err
+	}
+
+	// We only need these two parameters for SetDefaultPermissions
+	dto := &dashboards.SaveDashboardDTO{
+		User:  user,
+		OrgID: user.GetOrgID(),
+	}
+
+	// Temporary approach to set default permissions until we have a proper method in place via k8s
+	(*s.DashboardService).SetDefaultPermissions(ctx, dto, legacyDashboard, isProvisioned)
```

Comment:
Please double check that this is actually what we want. This is what we did in SaveDashboard for dashboards with ID == 0. 
https://github.com/grafana/grafana/blob/d7771f7d21f13a2061be8633b3033404c84f97af/pkg/services/dashboards/service/dashboard_service.go#L1025

Reply:
As long as this method is only called for new dashboard creation, this does look like what we want 👌

Reply:
Great, thanks for confirming